quickwit-doc-mapper/src/default_doc_mapper/date_time_type.rs

```diff
@@ -0,0 +1,539 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use chrono::TimeZone;
+use chrono_tz::Tz;
+use derivative::Derivative;
+use indexmap::IndexSet;
+use itertools::Itertools;
+use serde::ser::SerializeSeq;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use tantivy::{DatePrecision, DateTime};
+use time::format_description::well_known::{Iso8601, Rfc2822, Rfc3339};
+use time::OffsetDateTime;
+
+use super::default_as_true;
+
+/// A struct holding DateTime field options.
+#[derive(Clone, Serialize, Deserialize, Derivative)]
+#[derivative(Debug, Eq, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub struct QuickwitDateOptions {
+    #[serde(default)]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub description: Option<String>,
+
+    /// Accepted input formats.
+    #[serde(default = "default_input_formats")]
+    #[serde(serialize_with = "serialize_input_formats")]
+    #[serde(deserialize_with = "deserialize_input_formats")]
```

Comment:
`IndexSet` seems to serialize/deserialize without those custom serializer/deserializer. There should be no ordering issue for a set serialized as a list in JSON.